mypy/semanal.py

```diff
@@ -3873,8 +3910,10 @@ def add_symbol(self, name: str, node: SymbolTableNode,
             self.type.names[name] = node
         else:
             existing = self.globals.get(name)
-            if existing and (not isinstance(node.node, MypyFile) or
-                             existing.node != node.node) and existing.kind != UNBOUND_IMPORTED:
+            if (existing
+                    and (not isinstance(node.node, MypyFile) or existing.node != node.node)
+                    and existing.kind != UNBOUND_IMPORTED
```

Comment:
I would expect that the goal is to get rid of `UNBOUND_IMPORTED` and replace all such checks with `isinstance(node.node, ImportedName)`, what I am missing here?

Reply:
That's the end goal. This PR doesn't remove all instance of `UNBOUND_IMPORTED`. In particular, `import m` statements generate them still. I'll fix them in a separate PR.

Reply:
> I'll fix them in a separate PR.

OK, having smaller PRs makes sense.